editor/scene_tree_editor.cpp

```diff
@@ -993,6 +993,10 @@ void SceneTreeEditor::_renamed() {
 	}
 
 	if (new_name == n->get_name()) {
+		if (which->get_text(0).is_empty()) {
```

Comment:
This can be unconditional I think.

Reply:
Yeah empty `which->get_text(0)` is already handled above:
```cpp
String raw_new_name = which->get_text(0);
if (raw_new_name.strip_edges().is_empty()) {
```

Reply:
The reason I do that is:

There is the case that the user changes the name of this node in the editor to have the same name as its `class`, so the function basically just **return**.

But if it removes the name of this node and leaves it empty, the function must *revert* to the `class_name`

So the `new_name` here hold its `class_name`:
https://github.com/godotengine/godot/blob/98ed75f9e05f600b8cb5372df424ca3ae2f56ffe/editor/scene_tree_editor.cpp#L971-L981

but it does not classify which case, either because it is `empty` or because it has  `have the same name as its class`:
https://github.com/godotengine/godot/blob/98ed75f9e05f600b8cb5372df424ca3ae2f56ffe/editor/scene_tree_editor.cpp#L995-L1001

I cannot put the logic in this block code:
https://github.com/godotengine/godot/blob/98ed75f9e05f600b8cb5372df424ca3ae2f56ffe/editor/scene_tree_editor.cpp#L971-L972

because there are edge cases, where:
https://github.com/godotengine/godot/blob/98ed75f9e05f600b8cb5372df424ca3ae2f56ffe/editor/scene_tree_editor.cpp#L981
will verify and fix that.